### PR TITLE
Add DnParser#substring(begin, end).

### PR DESCRIPTION
**Pull request by dfish3r**

### CODE REVIEW

File: core/src/main/java/org/ldaptive/DnParser.java [hunk 1]
```diff
@@ -75,26 +75,67 @@ public static String getValue(final String dn, final String name)
...
+    }
+
+    final List<LdapAttribute> attrs = convertDnToAttributes(dn);
+    if (endIndex > attrs.size()) {
+      throw new IndexOutOfBoundsException("endIndex cannot be larger than the number of RDNs");
+    }
+
+    final StringBuilder sb = new StringBuilder();
+    for (int i = 0; i < attrs.size(); i++) {
+      if (i >= beginIndex && i < endIndex) {
```
Comment (serac) [inline core/src/main/java/org/ldaptive/DnParser.java]: Shouldn't the upper bound be `i <= endIndex`?

Comment (dfish3r) [inline core/src/main/java/org/ldaptive/DnParser.java]: Isn't that exactly the semantics that are enforced?
If not, provided a unit test to demonstrate.

Comment (serac) [inline core/src/main/java/org/ldaptive/DnParser.java]: You need to clarify the bounds on the parameters. I just checked String#substring, which works exactly like what you have done (including naming), so there's a solid precedent. You just need better javadocs on the acceptable parameter values for indices.

Comment (dfish3r) [inline core/src/main/java/org/ldaptive/DnParser.java]: Updated javadocs.

**Comment by serac** [comment]
Usage is clear now, thanks.

**Pull request merged**